plugins/common/opcua/input/input_client.go

```diff
@@ -356,8 +357,13 @@ func (o *OpcUAInputClient) UpdateNodeValue(nodeIdx int, d *ua.DataValue) {
 	}
 
 	if d.Value != nil {
-		o.LastReceivedData[nodeIdx].Value = d.Value.Value()
 		o.LastReceivedData[nodeIdx].DataType = d.Value.Type()
+
+		if o.LastReceivedData[nodeIdx].DataType == ua.TypeIDDateTime {
+			o.LastReceivedData[nodeIdx].Value = d.Value.Value().(time.Time).Format(time.RFC3339Nano)
+		} else {
+			o.LastReceivedData[nodeIdx].Value = d.Value.Value()
+		}
```

Comment:
Can we please check the type assertion as we will panic in case this is _not_ a time even though we expected it to be...
```suggestion
		o.LastReceivedData[nodeIdx].Value = d.Value.Value()
		if o.LastReceivedData[nodeIdx].DataType == ua.TypeIDDateTime {
			if t, ok := d.Value.Value().(time.Time); ok {
				o.LastReceivedData[nodeIdx].Value = t.Format(time.RFC3339Nano)
			}
		}
```